memtable/skiplistrep.cc

```diff
@@ -95,6 +95,37 @@ class SkipListRep : public MemTableRep {
     return (end_count >= start_count) ? (end_count - start_count) : 0;
   }
 
+  void UniqueRandomSample(const uint64_t& num_entries,
+                          const uint64_t& sample_size,
+                          std::unordered_set<const char*>* entries) override {
+    entries->clear();
+    SkipListRep::Iterator iter(&skip_list_);
+    // If the sample size is at least 25% of the total number of elements
+    // in the skip list, then simply iterate linearly over the list.
+    // (25% is arbitrary, could sample_size>N/log(N) if we assume iterating
+    // over each sample is O(N) and RandomSeek calls would be
+    // O(sample_size*log(N))).
+    if (sample_size >= (num_entries / 4)) {
+      Random rnd(1234);
+      int n = (sample_size > num_entries)
```

Comment:
The best pattern for this problem uses a dynamic threshold on a ranged random value. See https://cs.stackexchange.com/a/129757

Reply:
Perfect, this is exactly what was needed

Reply:
Actually, I am going for 10% of num_entries- the sample_size<<sqrt(num_entries) from the reference https://cs.stackexchange.com/a/129757 is not satifying for our situation (if N=1000, then sample_size of 2 would redirect to the first sampling algorithm and not the random picks...)